src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1090,61 +1087,117 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                     ? receiver.Type
                     : this.ContainingType;
                 method = new ErrorMethodSymbol(methodContainer, returnType, name);
-                args = BuildArgumentsForErrorRecovery(analyzedArguments);
             }
 
+            // until we've considered the extension methods, we don't rewrite the arguments.
+            args = extensionMethodsOfSameViabilityAreAvailable
+                ? analyzedArguments.Arguments.ToImmutable()
+                : BuildArgumentsForErrorRecovery(analyzedArguments, methods);
+
             var argNames = analyzedArguments.GetNames();
             var argRefKinds = analyzedArguments.RefKinds.ToImmutableOrNull();
             return BoundCall.ErrorCall(node, receiver, method, args, argNames, argRefKinds, isDelegate, invokedAsExtensionMethod: invokedAsExtensionMethod, originalMethods: methods, resultKind: resultKind);
         }
 
-        private static ImmutableArray<BoundExpression> BuildArgumentsForErrorRecovery(AnalyzedArguments analyzedArguments, ImmutableArray<ParameterSymbol> parameters)
+        private bool IsUnboundGeneric(MethodSymbol method)
         {
-            ArrayBuilder<BoundExpression> oldArguments = analyzedArguments.Arguments;
-            int argumentCount = oldArguments.Count;
-            int parameterCount = parameters.Length;
+            return method.IsGenericMethod && method.ConstructedFrom() == method;
```

Comment:
Reference comparisons are only valid for original symbols. `method.ConstructedFrom()` is not necessarily an original symbol. In short, no.